src/main/kotlin/org/radarbase/push/integration/garmin/backfill/GarminRequestGenerator.kt

```diff
@@ -89,6 +90,13 @@ class GarminRequestGenerator(
                         UserRoute(user.versionedId, route.toString()), startDate
                     ).coerceAtLeast(startDate)
                 }
+
+                if (startOffset <= backfillLimit) {
+                    // the start date is before the backfill limits
+                    logger.warn("Backfill limit exceeded for $user and $route. Resetting to earliest allowed start offset.")
+                    startOffset = backfillLimit.plus(Duration.ofDays(2))
```

Comment:
Btw, why do we need to add a plus 2 days here?

Reply:
Just added as a precaution (in case how they calculate the 2/5 years is)

Reply:
Alternatively, we can probably use `ZonedDateTime.now().minusYears(5).toInstant()` that will probably take into account leap years and timezone